File: platform_tools/android/apps/AndroidKit/src/main/java/org/skia/androidkit/Path.java
/*
 * Copyright 2021 Google Inc.
 *
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

package org.skia.androidkit;

public class Path {
    private long mNativeInstance;

    Path(long native_instance) {
        mNativeInstance = native_instance;
    }

    /**
     * Releases any resources associated with this Shader.
     */
    public void release() {
        nRelease(mNativeInstance);
        mNativeInstance = 0;
    }

    @Override
    protected void finalize() throws Throwable {
        release();
    }

    // package private
    long getNativeInstance() { return mNativeInstance; }

    private static native void nRelease(long nativeInstance);
}

File: modules/androidkit/src/Path.cpp
/*
 * Copyright 2021 Google Inc.
 *
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 */

#include <jni.h>

#include "include/core/SkPath.h"

namespace {
static void Path_Release(JNIEnv* env, jobject, jlong native_path) {
    delete reinterpret_cast<SkPath*>(native_path);
}

}  // namespace

int register_androidkit_Path(JNIEnv* env) {
    static const JNINativeMethod methods[] = {
        {"nRelease" , "(J)V"    , reinterpret_cast<void*>(Path_Release)},
    };

    const auto clazz = env->FindClass("org/skia/androidkit/Path");
    return clazz
        ? env->RegisterNatives(clazz, methods, SK_ARRAY_COUNT(methods))
        : JNI_ERR;
}
